src/compiler/checker.ts

```diff
@@ -27401,6 +27401,10 @@ namespace ts {
                 }
                 case SyntaxKind.NonNullExpression:
                     return getContextualType(parent as NonNullExpression, contextFlags);
+                case SyntaxKind.SatisfiesExpression:
+                    const assertedType = getTypeFromTypeNode((parent as SatisfiesExpression).type);
+                    const outerType = getContextualType(parent as SatisfiesExpression, contextFlags);

```

Comment:
After discussion in the original issue I think we want to just use the asserted type, so this can just be
```suggestion
                    return getTypeFromTypeNode((parent as SatisfiesExpression).type);
```

Reply:
Ok